clients/src/main/java/org/apache/kafka/clients/KafkaClient.java

```diff
@@ -81,6 +81,14 @@
     public List<ClientResponse> poll(long timeout, long now);
 
     /**
+     * Disconnect connection to a particular node (if there is one). The disconnection is asynchronous and will not be
+     * processed until the next {@link #poll(long, long) poll()} call.
```

Comment:
I see. I'll test the behaviour of the real selector just to be sure. In the likely case that it matches your description, I will do the following:
* Update `MockSelector`
* Update `NetworkClientTest`
* Update `Selector.disconnect` and `NetworkClient.disconnect` documentation
* Update implementation of `NetworkClient.disconnect` to update connection states

Have I covered everything?